cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java

```diff
@@ -614,7 +614,7 @@ public String getInputFormatClassName() {
   protected Collection<String> computeFilterInputPaths() {
     PartitionFilter filter;
     try {
-      filter = PartitionedFileSetArguments.getInputPartitionFilter(runtimeArguments, partitioning);
+      filter = PartitionedFileSetArguments.getInputPartitionFilter(runtimeArguments);
```

Comment:
If the partitionKey is:
{ key1: 1, key2: 3 }
and the filter configured in the runtime arguments is:
{ key1 == 1, key3 == 5 }

Then, the current behavior would deserialize the filter as:
{ key1 == 1 }   // (because key3 is not in the PARTITIONING), and then the key would match.

Wouldn't it be appropriate for `match` to return false, if there is a condition on a field that doesn't exist in the key?
Either:
1. `match` should return false.
2. the `PartitionedFileSetDataset` should validate the filter matches its partitioning after it deserializes it from the runtime arguments.

In this branch currently, just (1) will happen.